### ISSUE TITLE
Add event before hiding

### ISSUE THREAD

**Issue by leoquijano** [i-like-robots/EasyZoom#84]
Hi,

I'm using easyZoom for a special case where the zoom is enabled until a button is clicked, and disabled when another button is clicked. One way of implementing it would be with a PermaZoom option.

Another way, which I'm currently testing, is to add an event "beforeHide" before the zoom is hidden. Something like this:

```
// Callback function to execute before the flyout is removed.
onBeforeHide: $.noop,

// ... 

/**
 * Hide
 */
EasyZoom.prototype.hide = function() {
    if (!this.isOpen) return;
    **if (!this.opts.onBeforeHide.call(this)) return;**

     this.$flyout.detach();
     this.isOpen = false;

     this.opts.onHide.call(this);
 };
```

The custom event would then keep track of the current "open/closed" setting, and prevent the zoom from hiding if the setting is set to open.

Would this be something you would consider adding to the core library? I think it can be a useful feature.

**Comment by i-like-robots** [comment]
Thanks for your feedback @leoquijano. Has this been a useful feature?

I so, I think it would make sense to include the accompanying `onBeforeShow` callback. If you wish to submit a PR for the feature I'll be happy to include it.

**Comment by leoquijano** [comment]
I'm actively using it. PR is in #87. I added an additional beforeShow event, and also changed the naming a bit so it's only ```beforeShow``` and ```beforeHide```

**Comment by i-like-robots** [comment]
released 36890d75e66bbd34868aedc319179a82b03f7b70
